src/collection.jl

```diff
@@ -88,7 +71,7 @@ function fitscollection(basedir::String;
                 hdu isa ImageHDU || continue
                 header_data = read_header(hdu)
                 path = abspath ? Base.abspath(location) : location
-                name = keepext ? filename : first(split(filename, ext))[1:end-1]
+                name = keepext ? filename : first(split(filename, "." * ext))
```

Comment:
Does this work if the file has a funny name like `foo.fits.fits`?

Reply:
Doing this gives us only `foo`, I hope we don't get something strange like this. :joy:

Reply:
Well, don't expect users to choose sensible names :wink:

Reply:
I didn't try, but doesn't something like
```suggestion
                name = keepext ? filename : join(split(filename, ext)[1:end-1])
```
work?

Reply:
What would you suggest to tackle this case?

Reply:
Tried something like this:
```
ext = r"fits(\.tar\.gz)?"i
filename = "foo.fits.fits"

join(split(filename, "." * ext)[1:end-1])
```
This gives `foo` only

Reply:
```julia
parse_name(filename, ext::AbstractString, ::Val{false}) = first(rsplit(filename, "." * ext, limit=2))

function parse_name(filename, ext::Regex, ::Val{false})
	idxs = findall(ext, filename)
    return filename[1:first(last(idxs)) - 2]
end

parse_name(filename, ext, ::Val{true}) = filename

# within the parsing function
name = parse_name(filename, ext, Val(keepext))
```

Reply:
Should we modify the `ext` by adding `$` at the end and then just call `first(split(filename, "." ext))`

Reply:
>     return filename[1:first(last(idxs)) - 1]
Shouldn't this be `[1:first(last(idxs)) - 2]`?

Reply:
No, it should be 1 only.